shared/src/Twin.cs

```diff
@@ -48,6 +48,13 @@ public Twin(TwinProperties twinProperties)
         /// </summary>
         public string DeviceId { get; set; }
 
+        /// <summary>
+        /// The DTDL model id of the device.
+        /// The value will be null for a non-pnp device.
+        /// The value will be null for a pnp device until the device connects and registers with the model id.
+        /// </summary>
+        public string ModelId { get; set; }
+
```

Comment:
Is there an e2e possible for this scenario?

Reply:
It'll depend on if the hub's that support this are available in our canary region or not.

Reply:
Not yet. There is a task on my card to do it when their service is deployed in Canary.

Reply:
Based on Abhipsa's update below the service might be available today. So i'll add it in the next PR.